guides-markdown/multisig-wallet.md

```diff
@@ -0,0 +1,359 @@
+
+And then you can query by
+
+```sh
+curl $url/wallet/primary?token=${primarytoken}
+```
+
+### 2. Setting up hot multisig wallet for personal use
+
+Now let's create a multisig account in this `primary` wallet.
+
+TODO:
+
+### 3. Setting up a watch-only multisig wallet
+
+Here, we will create a multisig account as we did above with one difference.
+This time, the wallet is watch-only. So no private key will be on the server.
+
+Let's assume this wallet is for managing funds for three people.
+You, Alice, and Bob.
+
+> NOTE: Strictly speaking, there is no such a thing *watch-only account*.
+> Accounts will never hold private key no matter the wallet which it belongs to holds it's master private key or not.
+> If the wallet does, then it is able to derive account private key so the account will be *hot* .
+> In fact, this is the way how bcoin actually handles it's private key for an account, it derives every time when it's need.
+> If the wallet doesn't hold master private key (i.e. if it's watch-only), then an account derived from the wallet will be also watch-only.
+
+we can set watch-only wallet named `sharedWallet1` on bcoin node by
+
+```sh
+curl -X PUT $url/wallet/sharedWallet1 --data '
+  {"watchOnly": true, "type": "multisig", "m": 2, "n": 3 }
+  '
+
+# or by cli tool (installed with `npm i -g bclient`)
+bwallet-cli \
+  --url=${url} \
+  --network=testnet \
+  --api-key=mySecretApiKey \
+  --token=${myadmintoken} \ # Must use your admin token to create a new wallet
+  --watch-only=true \ # make it watch only
+  --type=multisig \
+  --m=2 \
+  --n=3 \
+  --name=multisig1 \
+  mkwallet sharedWallet1
+```
+
+> NOTE: this time, we specified `type` for wallet. But strictly speaking again, wallet itself does not hold property if it is multisig or not.
+> So some options we have passed here (i.e. `type`, `m`, `n`, and `name`) are for the default account created with the wallet.
+
+If the wallet is created successfully, than you will see the access token for that wallet in the return value.
+So let's set that token into variable named `sharedWalletToken` and pass to co-signers.
+
+You must register your public key to this account. Let's do it by using bcoin api.
+
+TODO: create and register key.
+
+```js
+
+```
+
+Next, cosigner Alice should create tpub too and set it to the account.
+Although it is possible that Alice can create it in the same way you did,
+lets assume that Alice uses [bitcoinjs-lib](https://github.com/bitcoinjs/bitcoinjs-lib) in her software.
+So the code will be something like ...
+
+```js
+const bjs = require("bitcoinjs-lib")
+const masterHD = new bjs.bip32.fromSeed(Buffer.from("ffffffffffffffffffffffffffffffff", "hex"), bjs.networks.testnet)
+// We are using segwit so following bip84
+const tpub = masterHD.derivePath("m/84'/0'/3'").neutered().toBase58()
+console.log(tpub)
+```
+
+and throw it to the bcoin server
+
+```sh
+curl -X PUT ${url}/wallet/sharedWallet1/account/multisig1\?token\=${sharedWalletToken} \
```

Comment:
Ah, correct endpoint was `/wallet/:id/shared-key` . I will fix.
I'd better write both case using `curl` and `bwallet-cli`. To me, sometimes it was more intuitive to use `curl` , since I only had to read source code for  `wallet/http.js` (instead of `bclient` , `bcurl` , and `wallet/http.js`)